### PR TITLE
BR-2122 phpcs update

### PR DESCRIPTION
**Pull request by kevinfodness**
* Update phpcs to the latest standards from wpcs and VIP.
* Move phpcs configuration to `phpcs.xml.dist`.
* Change method of phpcs installation to not require `composer.json`.
* Fix all errors and warnings reported by phpcs.

### CODE REVIEW

File: lib/class-sp-admin.php [hunk 1]
```diff
@@ -203,18 +221,38 @@ public function settings_page() {
...
+			case 'error': 
+				return __( 'Errors', 'searchpress' );
+			case 'warning': 
+				return __( 'Warnings', 'searchpress' );
+			case 'line': 
+				return __( 'Messages', 'searchpress' );
+			case 'success': 
+				return __( 'Success', 'searchpress' );
+			default:
+				return __( 'Unknown', 'searchpress' );
```
Comment (mboynes) [inline lib/class-sp-admin.php]: Looks like this addition breaks a failing phpunit test. I don't have a preference between updating the test or removing this default case, do whichever you prefer.

**Comment by benpbolton** [comment]
Travis failure may be the result of https://github.com/travis-ci/travis-ci/issues/9512

File: searchpress.php [hunk 2]
```diff
@@ -34,57 +34,57 @@
...
 }
 
 if ( defined( 'WP_CLI' ) && WP_CLI ) {
-	include SP_PLUGIN_DIR . '/bin/wp-cli.php';
+	include SP_PLUGIN_DIR . '/bin/wp-cli.php'; // phpcs:ignore WordPressVIPMinimum.Files.IncludingFile.IncludingFile
 }
 
 if ( defined( 'SP_DEBUG' ) && SP_DEBUG ) {
-	include SP_PLUGIN_DIR . '/lib/class-sp-debug.php';
+	include SP_PLUGIN_DIR . '/lib/class-sp-debug.php'; // phpcs:ignore WordPressVIPMinimum.Files.IncludingFile.IncludingFile
```
Comment (mboynes) [inline searchpress.php]: phpcs supports enable/disable like eslint. that would be better here, rather than repeating it like 20 times

Comment (kevinfodness) [reply searchpress.php]: At the file level? That makes sense. I'll do that.

Comment (mboynes) [reply searchpress.php]: I meant using `phpcs:disable WordPressVIPMinimum.Files.IncludingFile.IncludingFile` and `phpcs:enable WordPressVIPMinimum.Files.IncludingFile.IncludingFile`, but at the file level works too.

**Pull request merged**